tests/virt/node/general/test_machinetype.py

```diff
@@ -82,22 +84,39 @@ def migrated_vm(vm, machine_type_from_kubevirt_config):
     indirect=True,
 )
 def test_default_machine_type(machine_type_from_kubevirt_config, vm):
-    validate_machine_type(vm=vm, expected_machine_type=machine_type_from_kubevirt_config)
+    # Workaround for s390x (https://github.com/kubevirt/kubevirt/issues/14953), as machine type missing in config and
+    # hardcoded to s390_ccw_virtio in kubevirt code.
+    if machine_type_from_kubevirt_config == MachineTypesNames.s390_ccw_virtio:
+        expected_libvirt_machine_type = MachineTypesNames.s390_ccw_virtio_rhel9_6
+    else:
+        expected_libvirt_machine_type = machine_type_from_kubevirt_config
+    validate_machine_type(
+        vm=vm,
+        expected_machine_type=machine_type_from_kubevirt_config,
+        expected_libvirt_machine_type=expected_libvirt_machine_type,
+    )
 
 
 @pytest.mark.parametrize(
-    "vm, expected",
+    "vm, expected, expected_libvirt",
     [
         pytest.param(
             {"vm_name": "pc-q35", "machine_type": MachineTypesNames.pc_q35_rhel7_6},
             MachineTypesNames.pc_q35_rhel7_6,
-            marks=pytest.mark.polarion("CNV-3311"),
-        )
+            MachineTypesNames.pc_q35_rhel7_6,
+            marks=[pytest.mark.polarion("CNV-3311"), pytest.mark.x86_64()],
+        ),
+        pytest.param(
+            {"vm_name": "s390-ccw-virtio", "machine_type": MachineTypesNames.s390_ccw_virtio},
+            MachineTypesNames.s390_ccw_virtio,
+            MachineTypesNames.s390_ccw_virtio_rhel9_6,
+            marks=[pytest.mark.s390x()],
```

Comment:
architecture should not be used for test parameterization, if some config is different, please set it where needed.